### DISCUSSION TITLE
Mantine DataTable

### DISCUSSION THREAD

**Discussion by icflorescu** [mantinedev/mantine#2326] [Show]
I know there are already a few alternatives out there, but you might also want to take a look at [Mantine DataTable](https://icflorescu.github.io/mantine-datatable/), a component I've built with Mantine primitives (not React Table), that covers most of the features mentioned above and should support:

- Asynchronous data loading
- Pagination
- Multiple rows selection
- Scrollable behavior, while maintaining a fixed header, pagination footer and row selection column
- Column sorting
- Custom cell data rendering
- Row context-menu
- Dark themes

I've also tried to build a nice documentation website (following as much as possible the excellent Mantine's docs design patterns) and I've included a series of examples and usage scenarios.
The component properties are be self explanatory, well typed and also documented with JSDoc, so it should be quite straightforward to use.

[![Mantine DataTable](https://user-images.githubusercontent.com/581999/187723353-0e082dbc-925b-4d0d-a14e-94c00073ee08.png)](https://icflorescu.github.io/mantine-datatable/)

**Comment by rtivital** [comment]
Great job @icflorescu, thanks for sharing!

**Comment by icflorescu** [comment]
Thank you for your kind words and for building the best React UI framework, @rtivital!

**Reply by goodlythink** [reply]
The best datatable

**Comment by italodeandra** [comment]
Hey @icflorescu,

Realy nice DataTable, congrats 👏🏻. I'm already using it on my projects ❤️

**Comment by icflorescu** [comment]
Here's some exciting news:
> Starting with v.1.7.0, Mantine DataTable features [row expansion](https://icflorescu.github.io/mantine-datatable/examples/expanding-rows).

**Comment by icflorescu** [comment]
Thank you all for your kind words of encouragement, they really mean a lot to me!

Just a couple of points I'd like to reiterate here, after seeing and learning from people's reactions:

[Mantine DataTable](https://icflorescu.github.io/mantine-datatable/) is certainly not the single option in terms of data table / data grid functionality, and far from me to claim it's the best one out there. "The best" is always a subjective notion and it naturally depends on one's specific needs and preferences. Others offer features such as virtualization, better filtering support perhaps, etc., while mine comes with a few features that I considered to be important, such as:

- Context menu support;
- Additive / subtractive multiple row selection (when you're shift-clicking a row, the current selection is expanded / reduced naturally, as it happens in Gmail, for instance);
- Good UI integration with core Mantine components - *"as Mantine as it gets"*; this may be subjective, but IMO the component shouldn't feel out of place in your application UI;
- Lean footprint. There was a temporary shameful `peerDependency` (`lodash` :-P) in the initial versions, but I've managed to eliminate it with the help of a kind contributor.

There are, of course, a lot more features that could be added in time, and I'll hopefully manage to implement some of them with the help of other contributors - people who are kind and generous enough to contribute with quality code in their own time. For now, we have a lot of [discussions](https://github.com/icflorescu/mantine-datatable/discussions) going on regarding new feature requests; there are many ideas, but when it comes to open-source code contributions, time is always a scarce resource and implementing new features that work seamlessly in conjunction with existing ones isn't always trivial.

Thanks again for your support and encouragement!

![mantine-datatable](https://user-images.githubusercontent.com/581999/192305706-5198963b-507a-4dc7-9be1-e13bf6b1e9b0.png)

**Comment by mlshvdv** [comment]
@icflorescu Unbelievable, amazing work! Thanks!

**Comment by icflorescu** [comment]
Here's some good news:

### Mantine DataTable V7.1.0-alpha.5 is here 😅🎉

[![Mantine DataTable V7](https://github.com/icflorescu/mantine-datatable/assets/581999/0be62c66-978e-4ecc-9abe-0dad3254c94d)](https://icflorescu.github.io/mantine-datatable/)

Be advised that it comes with a few [breaking changes](https://github.com/icflorescu/mantine-datatable/blob/main/CHANGELOG.md#710-alpha1-2023-10-30), as expected.

The amount of effort spent to rewrite the entire documentation website with Next.js 14 was gruesome, but I had to do it in order to make sure the component works in every example.

If everything go as planned and we're not bumping into any critical bugs, I'm planning to release the final version in a day or two.

**Comment by ladderschool** [comment]
Is it possible to edit cells by navigating with arrow keys on the keyboard and if a cell is focused start typing in it?

**Comment by Akronae** [comment]
I'm really struggling with inputs in cells that trigger heavy re-renders & focus loss

**Comment by Kokleng-Dev** [comment]
is it work with nuxt or vue js?

**Reply by ryuhangyeong** [reply]
mantine is a `react` ui library.

Therefore, only `react` and `nextjs` are available.

suggestion `tanstack table vue`

https://tanstack.com/table/latest/docs/introduction

**Comment by BrendanC23** [comment]
I've had a lot of success using Mantine DataTable. It is both fully-featured and easy to use.

I also recently stumbled across https://v2.mantine-react-table.com/ (though I haven't had a chance to use it yet), which looks like a promising alternative. V2 is currently in beta. There's a comparison between both libraries [here](https://v2.mantine-react-table.com/#feature-comparison), though I can't speak to how accurate it is.
